src/lib/url.js

```diff
@@ -85,18 +142,10 @@ export const compileJsonApiUrl = ({
   fragment,
 }) => {
   const queryString = queryParams
-    .filter(
-      name =>
-        (query[name] && !Array.isArray(query[name])) || query[name].length,
-    )
-    .map(name => {
-      return name === 'fields'
-        ? Object.keys(query[name])
-            .map(type => `fields[${type}]=${[...query.fields[type]].join(',')}`)
-            .join('&')
-        : `${name}=${query[name].join(',')}`;
-    })
+    .filter(name => query[name] && !isEmpty(query[name]))
+    .map(name => compileQuery(name, query[name]))
```

Comment:
This is where I was thinking we could just pass `query` instead of `query[name]` to keep some symmetry with the `parse` functions.

Reply:
I think I see what you mean, at first I thought the problem was passing in `name` -but that's totally necessary to actually compile the query. I can change that arg, but then I just have to extract it inside the function instead. I figured since the map happened in the `compileJsonApiUrl` the current param value was already found.